hack/local-up-cluster.sh

```diff
@@ -710,6 +741,11 @@ function start_kubelet {
     fi
 
     mkdir -p "/var/lib/kubelet" &>/dev/null || sudo mkdir -p "/var/lib/kubelet"
+
+    if [[ ${EXTERNAL_CREDENTIAL_PROVIDERS} == "true" ]]; then
+      write_external_provider_config
```

Comment:
Seems like this should also be guarded by something like       
   if [[ "${CLOUD_PROVIDER:-}" == "aws" ]]; then
      write_aws_external_provider_config

We then have an easy path for other cloud providers to add their external provider to local up.

Reply:
Making this change.  We might want to make it a little smarter in the future so users can select multple providers, independent of their cloud provider.  I.e. some users might have multiple private registries, or fallbacks to other providers in case of outages.

Reply:
I would agree with that but at minimum providing lets provide an easy extensibility pattern. Others can add the azure/gce/vmware registries but we should make it clear where those changed should be made.